pkg/graph/resolvers/system_intake_status_requester.go

```diff
@@ -0,0 +1,107 @@
+		return calcSystemIntakeGRTMeetingStatusRequester(intake.GRTDate, currentTime), nil
+	case models.SystemIntakeStepFINALBIZCASE:
+		return calcSystemIntakeFinalBusinessCaseStatusRequester(intake.FinalBusinessCaseState)
+	case models.SystemIntakeStepGRBMEETING:
+		// this calc function doesn't use a switch statement and can't possibly return an error
+		return calcSystemIntakeGRBMeetingStatusRequester(intake.GRBDate, currentTime), nil
+	case models.SystemIntakeStepDECISION:
+		return calcSystemIntakeDecisionStatusRequester(intake.DecisionState)
+	default:
+		return "", fmt.Errorf("issue calculating the requester intake status, no valid step")
+	}
+}
+
+func calcSystemIntakeInitialFormStatusRequester(intakeFormState models.SystemIntakeFormState) (models.SystemIntakeStatusRequester, error) {
+	switch intakeFormState {
+	case models.SIRFSNotStarted:
+		return models.SISRInitialRequestFormNew, nil
+	case models.SIRFSInProgress:
+		return models.SISRInitialRequestFormInProgress, nil
+	case models.SIRFSEditsRequested:
+		return models.SISRInitialRequestFormEditsRequested, nil
+	case models.SIRFSSubmitted:
+		return models.SISRInitialRequestFormSubmitted, nil
+	default:
+		return "", fmt.Errorf("issue calculating the requester intake status, no valid intakeFormState")
+	}
+}
+
+func calcSystemIntakeDraftBusinessCaseStatusRequester(draftBusinessCaseState models.SystemIntakeFormState) (models.SystemIntakeStatusRequester, error) {
+	switch draftBusinessCaseState {
+	case models.SIRFSNotStarted, models.SIRFSInProgress:
+		return models.SISRDraftBusinessCaseInProgress, nil
+	case models.SIRFSEditsRequested:
+		return models.SISRDraftBusinessCaseEditsRequested, nil
+	case models.SIRFSSubmitted:
+		return models.SISRDraftBusinessCaseSubmitted, nil
+	default:
+		return "", fmt.Errorf("issue calculating the requester intake status, no valid draftBusinessCaseState")
+	}
+}
+
+func calcSystemIntakeGRTMeetingStatusRequester(grtDate *time.Time, currentTime time.Time) models.SystemIntakeStatusRequester {
+	if grtDate == nil || grtDate.After(currentTime) {
+		return models.SISRGrtMeetingReady
+	}
+
+	return models.SISRGrtMeetingAwaitingDecision
+}
+
+func calcSystemIntakeFinalBusinessCaseStatusRequester(finalBusinessCaseState models.SystemIntakeFormState) (models.SystemIntakeStatusRequester, error) {
+	switch finalBusinessCaseState {
+	case models.SIRFSNotStarted, models.SIRFSInProgress:
+		return models.SISRFinalBusinessCaseInProgress, nil
+	case models.SIRFSEditsRequested:
+		return models.SISRFinalBusinessCaseEditsRequested, nil
+	case models.SIRFSSubmitted:
+		return models.SISRFinalBusinessCaseSubmitted, nil
+	default:
+		return "", fmt.Errorf("issue calculating the requester intake status, no valid finalBusinessCaseState")
+	}
+}
+
+func calcSystemIntakeGRBMeetingStatusRequester(grbDate *time.Time, currentTime time.Time) models.SystemIntakeStatusRequester {
+	if grbDate == nil || grbDate.After(currentTime) {
```

Comment:
Echoing above, shouldn't there be a `READY_TO_SCHEDULE` status for a `nil` date?

Reply:
Not for overall intake status. ([Figma](https://www.figma.com/file/ChzAP34A2DVvQUNQwD7lCt/IT-Governance-Next?type=design&node-id=477-116633&mode=design&t=foWugeQRbZN35Xtx-0)) Task list status will be covered in EASI-2962.

Reply:
That's weird, but OK, if that's what the designs call for. Thanks for clarifying.